exporter/alertmanagerexporter/alertmanager_exporter.go

```diff
@@ -0,0 +1,248 @@
+	})
+	LabelMap["TraceID"] = model.LabelValue(event.traceID)
+	LabelMap["SpanID"] = model.LabelValue(event.spanID)
+	return LabelMap
+}
+
+func (s *alertmanagerExporter) convertEventstoAlertPayload(events []*alertmanagerEvent) []model.Alert {
+
+	var payload []model.Alert
+	for _, event := range events {
+		annotations := createAnnotations(event)
+
+		alert := model.Alert{
+			StartsAt:     time.Now(),
+			Labels:       model.LabelSet{"severity": model.LabelValue(event.severity), "event_name": model.LabelValue(event.spanEvent.Name())},
+			Annotations:  annotations,
+			GeneratorURL: s.generatorUrl,
+		}
+
+		payload = append(payload, alert)
+	}
+	return payload
+}
+
+func (s *alertmanagerExporter) postAlert(ctx context.Context, payload []model.Alert) error {
+	msg, _ := json.Marshal(payload)
+
+	req, err := http.NewRequestWithContext(ctx, "POST", s.endpoint, bytes.NewBuffer(msg))
+	if err != nil {
+		s.settings.Logger.Debug("error creating HTTP request", zap.Error(err))
+		return fmt.Errorf("error creating HTTP request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := s.client.Do(req)
+	if err != nil {
+		s.settings.Logger.Debug("error sending HTTP request", zap.Error(err))
+		return fmt.Errorf("error sending HTTP request: %w", err)
+	}
+
+	defer func() {
+		if closeErr := resp.Body.Close(); closeErr != nil {
+			s.settings.Logger.Warn("Failed to close response body", zap.Error(closeErr))
+		}
+	}()
+
+	_, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		s.settings.Logger.Debug("failed to read response body", zap.Error(err))
+		return fmt.Errorf("failed to read response body %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		s.settings.Logger.Debug("post request to Alertmanager failed", zap.Error(err))
+		return fmt.Errorf("request POST %s failed - %q", req.URL.String(), resp.Status)
+	}
+	return nil
+}
+
+func (s *alertmanagerExporter) pushTraces(ctx context.Context, td ptrace.Traces) error {
+
+	s.settings.Logger.Debug("TracesExporter", zap.Int("#spans", td.SpanCount()))
+
+	events := s.extractEvents(td)
+
+	if len(events) == 0 {
+		return nil
+	}
+
+	alert := s.convertEventstoAlertPayload(events)
+	err := s.postAlert(ctx, alert)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *alertmanagerExporter) start(_ context.Context, host component.Host) error {
+
+	client, err := s.config.HTTPClientSettings.ToClient(host, s.settings)
+	if err != nil {
+		s.settings.Logger.Error("failed to create HTTP Client", zap.Error(err))
+		return fmt.Errorf("failed to create HTTP Client: %w", err)
+	}
+	s.client = client
+	return nil
+}
+
+func (s *alertmanagerExporter) shutdown(context.Context) error {
+	return nil
+}
+
+func newAlertManagerExporter(cfg *Config, set component.TelemetrySettings) *alertmanagerExporter {
+
+	url := cfg.GeneratorURL
+
+	if len(url) == 0 {
+		url = "http://example.com/alert"
+	}
+
+	severity := cfg.DefaultSeverity
+
+	if len(severity) == 0 {
+		severity = "info"
```

Comment:
This should be a constant, defined at the top of the source file. It should also be part of the default config.